components/camel-grpc/src/main/java/org/apache/camel/component/grpc/GrpcUtils.java

```diff
@@ -35,6 +35,14 @@
 
     private GrpcUtils() {
     }
+    
+    public static String extractServiceName(String service) {
+        return service.substring(service.lastIndexOf(".") + 1);
+    }
+
+    public static String extractServicePackage(String service) {
+        return service.substring(0, service.lastIndexOf("."));
```

Comment:
If you don't include a package in the `.proto`, stubs and messages are put in the default package, so this may not work. You may want to add a simple test when fixing this (even if it's not a good practice to use the default package, it produces simpler URIs).